feat: install from package url

hypercall88 opened the pull request:
# feat: install clamav from package url

## 🗣 Description ##

This change adds the feature to install from a package URL instead of through the OS distribution package manger. The following default variables are added:

| Variable | Description | Default | Required |
|----------|-------------|---------|----------|
| clamav_install_from_package_manager | A boolean value to determine if the role should install from the system package manager. | `true` | No |
| clamav_package_version | The package version to install from the URL if not installing from the system package manager. | `1.3.1` | No |

In addition, a root-owned freshclam systemd service in `/etc/systemd/system` and freshclam configuration file at the specified `freshclam_configuration_path` are created via template files since the ClamAV daemon package no longer handles this. 

The `cron` packages are added to their own `dependencies` list to separate from the ClamAV-specific packages to install.

A molecule test, `package_install`, was added to test this new feature.

## 💭 Motivation and context ##

This feature allows playbook users to install the ClamAV package with bundled dependencies directly from the [ClamAV download URL](https://www.clamav.net/downloads).

This change is required on [RHEL version prior to 8.6](https://access.redhat.com/solutions/6981276) due to the epel repo no longer providing the required libjson-c.so.4 library. Rhel 8.4 is still a critical version in cloud Azure subscriptions.

## 🧪 Testing ##

I ran the pre-commit scripts and added a molecule test for the new feature. I ran all the molecule tests.

~I also updated the ansible and molecule versions in the test requirements, as it seems newer versions of molecule solved the bug mentioned in molecule:~
```
# See also cisagov/skeleton-packer#312.
ansible-core<2.16.3
# With the release of molecule v5 there were some breaking changes so
# we need to pin at v5 or newer. However, v5.0.0 had an internal
# dependency issue so we must use the bugfix release as the actual
# lower bound.
```

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

Commit:
feat: adds ability to install from package URL instead of package manager

Commit:
chore: pre-commit fixes

Commit:
chore: update README

Commit:
fix: create freshclam service before conf modifications

Commit:
fix: remove notify clamd with package install since daemon is not setup as service

jsf9k commented:
I will rebase this PR once #79 is merged.

jsf9k commented:
> I will rebase this PR once #79 is merged.

@desebjohnston - I rebased, but I would need to force-push the changes to your branch.  Can I do that?  I did remove the changes you made to `requirements-test.txt` and `.pre-commit-config.yaml` since (1) I don't think they are necessary now that #79 has been merged and (2) those changes would have to be made to [cisagov/skeleton-ansible-role](https://github.com/cisagov/skeleton-ansible-role) and then allowed to flow down to that repo's descendants via [cisagov/action-lineage](https://github.com/cisagov/action-lineage).

hypercall88 commented:
> > I will rebase this PR once #79 is merged.
> 
> @desebjohnston - I rebased, but I would need to force-push the changes to your branch. Can I do that? I did remove the changes you made to `requirements-test.txt` and `.pre-commit-config.yaml` since (1) I don't think they are necessary now that #79 has been merged and (2) those changes would have to be made to [cisagov/skeleton-ansible-role](https://github.com/cisagov/skeleton-ansible-role) and then allowed to flow down to that repo's descendants via [cisagov/action-lineage](https://github.com/cisagov/action-lineage).

Yes, that's absolutely fine. Thanks for consuming my changes!

jsf9k commented:
> Thanks for consuming my changes!

Thank _you_ for the contribution!

Commit:
chore: remove deprecated ansible.builtin.yum module
chore: renamed test_packages_and_services to non-conflicting name for mypy
chore: changed test_files_and_dirs.py to symbolic link to default test
fix: fixed virus scan template to explicitly reference freshclam database directory, otherwise clamscan installed from package will look in the wrong directory"

Commit:
fix: fix pathing for package installed clamscan in virus_scan cronjob

hypercall88 commented:
FYSA I have a few fixes I had to push after inspecting my logs and realizing the scan cronjob was failing this morning due to pathing issues.

I also fixed a couple linting issues.

jsf9k commented:
@desebjohnston - The `modify_config` checks are failing because GH Actions has been hitting `freshclam` too hard and some of the IPs it uses have been blocked.  I'll handle those failures so you can focus on any failing `package_install` checks that are actually relevant to this PR.

jsf9k committed:
Integrate upstream changes to default scenario into package_install scenario

jsf9k committed:
Fix typos

jsf9k committed:
Remove ARM64 support for the package_install scenario

The packages that ClamAV offers directly on its website do not support
the ARM64 architecture.  See https://www.clamav.net/downloads.

jsf9k committed:
Remove unnecessary quotes

jsf9k committed:
Alphabetize Ansible task attributes

jsf9k committed:
Remove vars file for Fedora 38

We no longer support Fedora 38.

jsf9k committed:
Clean up language and remove unnecessary space

jsf9k commented:
The remaining builds that are failing are doing so [for the same reason that they did before](https://github.com/cisagov/ansible-role-clamav/pull/70#issuecomment-1821645724).  The issue is that some GH Actions IPs are being blocked by ClamAV.

jsf9k reviewed (approved):
Thank you for the contribution, @desebjohnston!

dav3r commented on tasks/main.yml:
```diff
@@ -11,10 +11,22 @@
+    state: present
+  when: clamav_install_from_package_manager
+
+- name: Install ClamAV from package url
```
Minor capitalization quibble:
```suggestion
- name: Install ClamAV from package URL
```

dav3r reviewed (approved):
Looks great, thank you for contributing! 🚀

jsf9k committed:
Capitalize the word "URL"

Co-authored-by: dav3r <daver@geekpad.com>
